app/serializers/cache/additional_code_serializer.rb

```diff
@@ -26,21 +20,35 @@ def as_json
         validity_start_date: additional_code.validity_start_date,
         validity_end_date: additional_code.validity_end_date,
         measure_ids: measures.map(&:measure_sid),
-        measures: measures.map do |measure|
-          {
-            id: measure.measure_sid,
-            measure_sid: measure.measure_sid,
-            validity_start_date: measure.validity_start_date,
-            validity_end_date: measure.validity_end_date,
-            goods_nomenclature_item_id: measure.goods_nomenclature_item_id,
-            goods_nomenclature_sid: measure.goods_nomenclature_sid,
-            goods_nomenclature_id: measure.goods_nomenclature_sid,
-            goods_nomenclature: goods_nomenclature_attributes(measure.goods_nomenclature),
-            geographical_area_id: measure.geographical_area_id,
-            geographical_area: geographical_area_attributes(measure.geographical_area),
-          }
-        end,
+        measures: serialized_measures,
       }
     end
+
+    private
+
+    def serialized_measures
+      measures.map do |measure|
+        {
+          id: measure.measure_sid,
+          measure_sid: measure.measure_sid,
+          validity_start_date: measure.validity_start_date,
+          validity_end_date: measure.validity_end_date,
+          goods_nomenclature_item_id: measure.goods_nomenclature_item_id,
+          goods_nomenclature_sid: measure.goods_nomenclature_sid,
+          goods_nomenclature_id: measure.goods_nomenclature_sid,
+          goods_nomenclature: goods_nomenclature_attributes(measure.goods_nomenclature),
+          geographical_area_id: measure.geographical_area_id,
+          geographical_area: geographical_area_attributes(measure.geographical_area),
+        }
+      end
+    end
+
+    def measures
+      additional_code
+        .measures_dataset.eager(:goods_nomenclature)
```

Comment:
minor but can the eager be on its own line - makes it easier to spot in the future